malfeasance/handler_test.go

```diff
@@ -728,3 +729,57 @@ func TestHandler_HandleMalfeasanceProof_validateHare(t *testing.T) {
 		require.Equal(t, pubsub.ValidationAccept, h.HandleMalfeasanceProof(context.Background(), "peer", data))
 	})
 }
+
+func TestDataAndTypeConsistent(t *testing.T) {
+	signer, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	target := 10
+	m1 := types.BallotMetadata{
+		Layer:   types.NewLayerID(uint32(target)),
+		MsgHash: types.Hash32{1, 1, 1},
+	}
+	m2 := types.ATXMetadata{
+		Target:  types.EpochID(target),
+		MsgHash: types.Hash32{2, 2, 2},
+	}
+	m1buf, err := codec.Encode(&m1)
+	require.NoError(t, err)
+	final1 := make([]byte, 1+len(m1buf))
+	final1[0] = types.MultipleBallots
+	copy(final1[1:], m1buf)
+
+	m2buf, err := codec.Encode(&m2)
+	require.NoError(t, err)
+	final2 := make([]byte, 1+len(m2buf))
+	final2[0] = types.MultipleATXs
+	copy(final2[1:], m2buf)
+
+	db := sql.InMemory()
+	lg := logtest.New(t)
+	mcp := malfeasance.NewMockconsensusProtocol(gomock.NewController(t))
+	pke, err := signing.NewPubKeyExtractor()
+	require.NoError(t, err)
+
+	createIdentity(t, db, signer)
+	h := malfeasance.NewHandler(datastore.NewCachedDB(db, lg), lg, "self", mcp, pke)
+	msg, err := codec.Encode(&types.MalfeasanceGossip{
+		MalfeasanceProof: types.MalfeasanceProof{
+			Proof: types.Proof{
+				Type: types.MultipleBallots,
+				Data: &types.BallotProof{
+					Messages: [2]types.BallotProofMsg{
+						{InnerMsg: m1, Signature: signer.Sign(final1)},
+						{InnerMsg: *(*types.BallotMetadata)(unsafe.Pointer(&m2)), Signature: signer.Sign(final2)},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	require.Equal(t, pubsub.ValidationIgnore, h.HandleMalfeasanceProof(context.Background(), "", msg))
```

Comment:
@dshulyak this is essentially the test you supplied (thank you!)
it's not rejected because the code failed looking up the node ID after extracting the public key. so ignore instead